src/main/java/com/icthh/xm/ms/entity/lep/ElasticIndexManager.java

```diff
@@ -0,0 +1,70 @@
+package com.icthh.xm.ms.entity.lep;
+
+import com.icthh.xm.ms.entity.domain.XmEntity;
+import com.icthh.xm.ms.entity.repository.search.XmEntitySearchRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.support.TransactionSynchronizationAdapter;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.springframework.transaction.support.TransactionSynchronizationManager.registerSynchronization;
+
+@Slf4j
+@RequiredArgsConstructor
+@TransactionScoped
+@Component
+public class ElasticIndexManager {
+
+    private final XmEntitySearchRepository searchRepository;
+    private final List<XmEntity> entityToSave = new ArrayList<>();
+    private final List<XmEntity> entityToDelete = new ArrayList<>();
+
+    public void addEntityToSave(XmEntity entity) {
+        if (entityToSave.isEmpty()) {
+            registerSynchronization(new TransactionSynchronizationAdapter() {
+                @Override
+                public void afterCompletion(int status) {
+                    super.afterCompletion(status);
+                    if (status == STATUS_COMMITTED) {
+                        try {
+                            if (!entityToSave.isEmpty()) {
+                                searchRepository.saveAll(entityToSave);
+                            }
+                        } catch (Exception e) {
+                            log.error("Error in afterCompletion stage during transaction synchronization for entity: {}",
```

Comment:
@sergeysenja1992 seems this log will print a lot of entities. is not it? Do we really need to do it?